### PR TITLE
GEOPY-501: Allow to run MT inversion with single component

### PR DESCRIPTION
**Pull request by benk-mira**
**GEOPY-501 - Allow to run MT inversion with single component**
Handles bugs when one channel selected.

### CODE REVIEW

File: geoapps/inversion/components/factories/survey_factory.py [hunk 1]
```diff
@@ -150,56 +150,52 @@ def build(
...
-
-        if self.factory_type == "magnetotellurics":
-            local_data = data.observed[component_map[component]][channel][local_index]
-            local_uncertainties = data.uncertainties[component_map[component]][channel][
-                local_index
-            ]
-        elif self.factory_type == "tipper":
-            local_data = data.observed[component][channel][local_index]
-            local_uncertainties = data.uncertainties[component][channel][local_index]
+    def _something(self, data, channel, local_index):
```
Comment (domfournier) [inline geoapps/inversion/components/factories/survey_factory.py:153]: Need to rename this better

File: geoapps/inversion/components/factories/survey_factory.py [hunk 2]
```diff
@@ -150,56 +150,52 @@ def build(
...
-
-                    (
-                        local_data["_".join([str(channel), str(comp)])],
-                        local_uncertainties["_".join([str(channel), str(comp)])],
-                    ) = self._localize(data, comp, channel, local_index)
+                dat, unc = self._something(data, channel, local_index)
+                local_data.update(dat)
+                local_uncertainties.update(unc)
 
             data_vec = self._stack_channels(local_data, "cluster_components")
```
Comment (domfournier) [inline geoapps/inversion/components/factories/survey_factory.py:200]: Can we rename "cluster_locs" too?

Comment (domfournier) [inline geoapps/inversion/components/factories/survey_factory.py:200]: Can we rename "cluster_locs" too?

**Review by domfournier — APPROVED** [review]

File: geoapps/inversion/components/factories/survey_factory.py (hunk 1 above)
Comment (benk-mira) [reply geoapps/inversion/components/factories/survey_factory.py:153]: done

File: geoapps/inversion/components/factories/survey_factory.py (hunk 2 above)
Comment (benk-mira) [reply geoapps/inversion/components/factories/survey_factory.py:200]: done

Comment (benk-mira) [reply geoapps/inversion/components/factories/survey_factory.py:200]: done

**Review by domfournier — DISMISSED** [review]

**Pull request merged**